homeassistant/components/netgear/device_tracker.py

```diff
@@ -39,132 +42,158 @@
-    def __init__(
-        self,
-        api,
-        devices,
-        excluded_devices,
-        accesspoints,
-    ):
-        """Initialize the scanner."""
-        self.tracked_devices = devices
-        self.excluded_devices = excluded_devices
-        self.tracked_accesspoints = accesspoints
-        self.last_results = []
-        self._api = api
-
-    def scan_devices(self):
-        """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
-
-        devices = []
-
-        for dev in self.last_results:
-            tracked = (
-                not self.tracked_devices
-                or dev.mac in self.tracked_devices
-                or dev.name in self.tracked_devices
-            )
-            tracked = tracked and (
-                not self.excluded_devices
-                or not (
-                    dev.mac in self.excluded_devices
-                    or dev.name in self.excluded_devices
-                )
-            )
-            if tracked:
-                devices.append(dev.mac)
-                if (
-                    self.tracked_accesspoints
-                    and dev.conn_ap_mac in self.tracked_accesspoints
-                ):
-                    devices.append(f"{dev.mac}_{dev.conn_ap_mac}")
+async def async_get_scanner(hass, config):
+    """Import Netgear configuration from YAML."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-        return devices
+    _LOGGER.warning(
+        "Your Netgear configuration has been imported into the UI, "
+        "please remove it from configuration.yaml. "
+        "Loading Netgear via platform setup is now deprecated"
+    )
 
-    def get_device_name(self, device):
-        """Return the name of the given device or the MAC if we don't know."""
-        parts = device.split("_")
-        mac = parts[0]
-        ap_mac = None
-        if len(parts) > 1:
-            ap_mac = parts[1]
-
-        name = None
-        for dev in self.last_results:
-            if dev.mac == mac:
-                name = dev.name
-                break
+    return None
 
-        if not name or name == "--":
-            name = mac
 
-        if ap_mac:
-            ap_name = "Router"
-            for dev in self.last_results:
-                if dev.mac == ap_mac:
-                    ap_name = dev.name
-                    break
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Netgear component."""
+    router = hass.data[DOMAIN][entry.unique_id]
+    tracked = set()
 
-            return f"{name} on {ap_name}"
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        add_entities(router, async_add_entities, tracked)
 
-        return name
+    router.listeners.append(
```

Comment:
We can drop `router.listeners` and use `entry.async_on_unload`

Reply:
thanks